src/Accounts/Accounts/CommonModule/ContextAdapter.cs

```diff
@@ -115,8 +117,7 @@ public void OnNewRequest(InvocationInfo invocationInfo, string correlationId, st
                 {
                     endpointResourceIdKey = endpointResourceIdKey ?? AzureEnvironment.Endpoint.ResourceManager;
                     var context = GetDefaultContext(_provider, invocationInfo);
-                    await AuthorizeRequest(context, request, cancelToken, endpointResourceIdKey, endpointSuffixKey, tokenAudienceConverter);
-                    return await next(request, cancelToken, cancelAction, signal);
```

Comment:
As we disscussed in today's meeting, we would like to revisit the part of the design related to autorest -- instead of adding CAE logic to the `AddAuthorizeRequestHandler` method, which has little to do with CAE, it seems better to spilt that into a new handler specially for CAE.

However, redoing authentication requires information about the access token (such as environment, audience). If the logic is departed from authentication, how does it get those info?

A second concern is about calling `next()`. We want to retry the failed request by calling `next()` for the second time (line 211), however if there are more handlers after this one, those handlers will be called twice against the same http request, which may have side-effect such as duplicated HTTP header. Is there a way allowing us to resend the request without duplicately calling the handlers?

Reply:
> As we disscussed in today's meeting, we would like to revisit the part of the design related to autorest -- instead of adding CAE logic to the `AddAuthorizeRequestHandler` method, which has little to do with CAE, it seems better to spilt that into a new handler specially for CAE.
> 
> However, redoing authentication requires information about the access token (such as environment, audience). If the logic is departed from authentication, how does it get those info?
> 
> A second concern is about calling `next()`. We want to retry the failed request by calling `next()` for the second time (line 211), however if there are more handlers after this one, those handlers will be called twice against the same http request, which may have side-effect such as duplicated HTTP header. Is there a way allowing us to resend the request without duplicately calling the handlers?


Thanks for @isra-fel's conclusions. After the discussion with @dolauli,  we find a way to seperate the authentication and reauthentication steps. However, the codes of these 2 parts are coupled, we decide to include them into one step currently.
@dolauli Could you help to review the related codes?

Thanks